### PR TITLE
Add new endpoint and attchment support of gcnotify.

### PR DESCRIPTION
**Pull request by pwei1018**
*Issue #:* 
[/bcgov/entity/13966](https://github.com/bcgov/entity/issues/13966)
[/bcgov/entity/14257](https://github.com/bcgov/entity/issues/14257)

*Description of changes:*
1. added a new endpoint to validate email address with email-validator library and millionverifier api;
2. support attachment on GC Notify Service;
3. update github actions versions;

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by pwei1018** [comment]
I will fix the unit test later.

**Review by ozamani9gh — APPROVED** [review]
dont fully understand all of this code, but it LOOKS GOOD and Thorough

**Review by eve-git — APPROVED** [review]

File: notify-api/src/notify_api/utils/enums.py [hunk 1]
```diff
@@ -24,3 +24,14 @@ class Role(BaseEnum):
     SMS = 'sms'
     JOB = 'job'
     INVALID = 'invalid'
+
+
+class MillionverifierResult(BaseEnum):
+    """The result of email verification."""
+
+    OK = 'ok'
```
Comment (thorwolpert) [inline notify-api/src/notify_api/utils/enums.py:32]: you can use 
```suggestion
    OK = auto()
```
If you look at the sql enum implementation you'll see that it also sets up auto to create a lowercase name of the ENUM.

**Review by thorwolpert — COMMENTED** [review]
I like the use of pydantic, we should use that more.

1 suggestion on changing the enum setup to use auto & the automatic creation of the enum value.

**Pull request merged**